acceptance/tests/api-gateway_v2/api_gateway_v2_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package apigateway_v2
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+
+	"github.com/hashicorp/consul-k8s/acceptance/framework/consul"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/helpers"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/k8s"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/logger"
+	meshv2beta1 "github.com/hashicorp/consul-k8s/control-plane/api/mesh/v2beta1"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Test that api gateway basic functionality works in a default installation and a secure installation for V2
+func TestAPIGateway_V2_Basic(t *testing.T) {
```

Comment:
should we have an env var check here and skip if it's not set/empty?

Reply:
When I tested the env check in the [main](https://github.com/hashicorp/consul-k8s/pull/3540/files#diff-73333c5946f976c570d1673d58af6ed82da06fa4cdb446c5680901c1ac862d21R17) was sufficient for the entire package to be skipped.